pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go

```diff
@@ -317,7 +317,10 @@ func (e *AzureMonitorDatasource) parseResponse(amr types.AzureMonitorResponse, q
 				Unit: toGrafanaUnit(amr.Value[0].Unit),
 			})
 		}
-		resourceID := labels["microsoft.resourceid"]
+		resourceID, ok := labels["microsoft.resourceid"]
+		if !ok {
+			resourceID = labels["Microsoft.ResourceId"]
```

Comment:
Are we falling back on this because it's always defined?

Reply:
It's not always defined. I found out that sometimes this label is `"Microsoft.ResourceId"` but sometimes is not set (for the single resource use case). If not set, this fallsback to getting the name and resourceID from the query URL.